CMakeLists.txt

```diff
@@ -262,8 +262,8 @@ if(BUILD_DEV)
 else()
     set(MIOPEN_BUILD_DEV 0)
     set(MIOPEN_SYSTEM_DB_PATH "${CMAKE_INSTALL_PREFIX}/${DATA_INSTALL_DIR}/db" CACHE PATH "Default path of system db files")
-    set(MIOPEN_USER_DB_PATH "~/.config/miopen/" CACHE STRING "Default path of user db files")
-    set(MIOPEN_CACHE_DIR "~/.cache/miopen/" CACHE STRING "")
+    set(MIOPEN_USER_DB_PATH "~/.config/miopen.${MIOpen_VERSION_MAJOR}_${MIOpen_VERSION_MINOR}_${MIOpen_VERSION_PATCH}_${MIOpen_VERSION_TWEAK}/" CACHE STRING "Default path of user db files")
```

Comment:
This will result in `~/.cache/miopen.M.m.p.t/M.m.p/miopen.udb` I think. Example: `~/.cache/miopen.2.4.0.439824/2.4.0/miopen.udb`.
Don't we instead want:
`~/.cache/miopen/M.m.p.t/miopen.udb`
For example:
`~/.cache/miopen/2.4.0.439824/miopen.udb`

Reply:
This will also move user find-db files from `~/.config/` to "versioned" directory, but these files already contains version in db filenames: `~/.config/miopen/2.4.0.439824/gfx906_60.HIP.2_4_0.ufdb.txt`, so we are doing the same job twice.

The idea for perf-db and find-db was to use paths to denote base directory, and encode whatever is necessary (for differentiating databases) into filenames.

The filesystem-based binary cache uses subdirectories, so the "separating" information is encoded into the name of the topmost directory. Now we do not use subdirs, but the directory remains the same.

(Current implementation is messed up :face_with_head_bandage:  Perhaps it's time to make it uniform, but I still do not have clear idea.)